crates/ra_assists/src/handlers/add_function.rs

```diff
@@ -0,0 +1,788 @@
+use ra_syntax::{
+    ast::{self, AstNode},
+    SyntaxKind, SyntaxNode, TextUnit,
+};
+
+use crate::{Assist, AssistCtx, AssistId};
+use ast::{edit::IndentLevel, ArgListOwner, CallExpr, Expr};
+use hir::HirDisplay;
+use rustc_hash::{FxHashMap, FxHashSet};
+
+// Assist: add_function
+//
+// Adds a stub function with a signature matching the function under the cursor.
+//
+// ```
+// struct Baz;
+// fn baz() -> Baz { Baz }
+// fn foo() {
+//      bar<|>("", baz());
+// }
+//
+// ```
+// ->
+// ```
+// struct Baz;
+// fn baz() -> Baz { Baz }
+// fn foo() {
+//      bar("", baz());
+// }
+//
+// fn bar(arg: &str, baz: Baz) {
+//     unimplemented!()
+// }
+//
+// ```
+pub(crate) fn add_function(ctx: AssistCtx) -> Option<Assist> {
+    let path_expr: ast::PathExpr = ctx.find_node_at_offset()?;
+    let call = path_expr.syntax().parent().and_then(ast::CallExpr::cast)?;
+    let path = path_expr.path()?;
+
+    if path.qualifier().is_some() {
```

Comment:
we could allow adding functions to other modules in the future as well :thinking: